### PR TITLE
FFI translation interface

### PR DESCRIPTION
**Pull request by uselessgoddess**

### CODE REVIEW

File: csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+
+    public class UnitedMemoryLinks<TLink> : DisposableBase, ILinks<TLink>
+    {
+        private static readonly UncheckedConverter<Byte, TLink> from_u8 = UncheckedConverter<Byte, TLink>.Default;
+        private static readonly UncheckedConverter<UInt16, TLink> from_u16 = UncheckedConverter<UInt16, TLink>.Default;
+        private static readonly UncheckedConverter<UInt32, TLink> from_u32 = UncheckedConverter<UInt32, TLink>.Default;
+        private static readonly UncheckedConverter<UInt64, TLink> from_u64 = UncheckedConverter<UInt64, TLink>.Default;
+        private static readonly UncheckedConverter<TLink, UInt64> from_t = UncheckedConverter<TLink, UInt64>.Default;
+
+        public LinksConstants<TLink> Constants { get; }
```
Comment (FreePhoenix888) [inline csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs:143]: Why not readonly field?

File: csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+                            break;
+                        case UInt64:
+                            t64:
+                            Methods.UInt64UnitedMemoryLinks_Drop(_body);
+                            break;
+                        default:
+                            throw new NotImplementedException();
+                    }
+                }
+            }
```
Comment (FreePhoenix888) [inline csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs:422]:
```suggestion
            if (wasDisposed) 
            {
            return;
            }
            TLink t = default;
            unsafe
            {
                switch (t)
                {
                    case Byte t8:
                        Methods.ByteUnitedMemoryLinks_Drop(_body);
                        break;
                    case UInt16 t16:
                        Methods.UInt16UnitedMemoryLinks_Drop(_body);
                        break;
                    case UInt32 t32:
                        Methods.UInt32UnitedMemoryLinks_Drop(_body);
                        break;
                    case UInt64:
                        t64:
                        Methods.UInt64UnitedMemoryLinks_Drop(_body);
                        break;
                    default:
                        throw new NotImplementedException();
                }
            }
```

**Review by FreePhoenix888 — APPROVED** [review]

File: csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs (hunk 1 above)
Comment (uselessgoddess) [reply csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs:143]: Sorry bro, it's interface

File: csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs (hunk 2 above)
Comment (uselessgoddess) [reply csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs:422]: ### Why?

File: csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs [hunk 3]
```diff
@@ -0,0 +1,425 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using Platform.Converters;
+using Platform.Disposables;
+using IDisposable = Platform.Disposables.IDisposable;
```
[the remark this answers is not in the document]
Comment (uselessgoddess) [reply csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs:6]: Yeap

[the remark this answers is not in the document]
Comment (FreePhoenix888) [reply csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs:6]: @uselessgoddess  Where?

File: csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs (hunk 2 above)
Comment (FreePhoenix888) [reply csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs:422]: More readable, less indents

File: csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (uselessgoddess) [reply csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs:6]: Nowhere

**Pull request merged**